packages/commonjs/src/helpers.js

```diff
@@ -35,8 +35,14 @@ export function getDefaultExportFromNamespaceIfNotNamed (n) {
 }
 
 export function getAugmentedNamespace(n) {
-	if (n.__esModule) return n;
-	var a = Object.defineProperty({}, '__esModule', {value: true});
+  var f = n.default;
+	if (typeof f == "function") {
+		var a = function () {
+			return f.apply(this, arguments);
+		};
+		a.prototype = f.prototype;
+  } else a = {};
+  Object.defineProperty(a, '__esModule', {value: true});
 	Object.keys(n).forEach(function (k) {
 		var d = Object.getOwnPropertyDescriptor(n, k);
 		Object.defineProperty(a, k, d.get ? d : {
```

Comment:
@lukastaegert sorry for tagging you directly. I have an issue in upgrading from `"@rollup/plugin-commonjs"@21.1.0` to `"@rollup/plugin-commonjs"@latest`. I tracked problem down to version `"@rollup/plugin-commonjs"@22.0.0` and it looks like all changes in this version are in this PR. I hope you'll be able to provide valuable insights only by reading my question, that's why I tagged you.

I have a case when `n` which arrives to `getAugmentedNamespace` function already contains `__esModule` property. 
As the `if (n.__esModule) return n;` line was removed by this PR, the code now throws at the `Object.defineProperty(a, k, d.get ? d : {` line because it tries to define `__esModule` property again.

Unfortunately my codebase is complex and I didn't managed to create a reproducible repo for opening an issue. Looking into exports of `n` it looks similar to `react-dom` (we use v17.0.2), but that's all I was able to find. Can you please tell me in which circumstances `getAugmentedNamespace` helper is added to the bundle? That may help with creating a repo to reproduce and better understanding of root cause.

How do you feel about adding back the `if (n.__esModule) return n;` line or modifying `Object.keys(n).forEach(function (k) {` line to be `Object.keys(n).filter(k => k !== '__esModule').forEach(function (k) {` ?

Thanks!

Reply:
It is actually described in the readme https://github.com/rollup/plugins/blob/34f4c4f294b1c024552fc9674954a2d6b978e360/packages/commonjs/README.md#requirereturnsdefault in the `requireReturnsDefault` section, though the helper code there is not the most current one. It is only used when `requireReturnsDefault` is set to `false` or not set. It is used when importing an ES module, or when importing a module where the type cannot be determined.
But yes, that is an oversight. I guess the motivation was that `__esModule` is a valid ES export that would however then return a namespace object without a prototype while `require` usually returns an object that has the `Object.prototype`.
A solution could be to make `__esModule` configurable. One would need to look into old test cases to see if adding the `__esModule` short cut back breaks anything important.

Reply:
Thank you very much! That was super helpful. 
Adding `requireReturnsDefault: 'namespace'` seems to be solving issue in my case. `react-dom` is external in my case, so I guess this is actually what we want.
I'll try to add back a `__esModule` shortcut and see if PR will be green.

Reply:
@lukastaegert unfortunately adding `requireReturnsDefault` option fixes the code for `react-dom` but resulted in a bad code generated to bundle some other NPM package. After a number of tries I think adding check back will be a much better fix.
I opened a MR - https://github.com/rollup/plugins/pull/1379. Only snapshot tests failed, and I reviewed them one by one and the difference was only in the line which I added back to helper. Which makes me think I didn't break something.
I'll be grateful for your review. Thanks!